src/main/java/edu/wpi/grip/ui/OperationView.java

```diff
@@ -88,5 +88,28 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
             Step step = new Step(this.eventBus, this.operation);
             this.eventBus.post(new StepAddedEvent(step));
         });
+
+
+        // Ensures that when this element is hidden that it also removes its size calculations
+        this.managedProperty().bind(this.visibleProperty());
+    }
+
+
+    /**
+     * Uses the search string from the palette and allows this element to determine if it should be visible
+     *
+     * @param searchString The search string that should be used to determine visibility
+     */
+    public void determineVisibility(final String searchString) {
```

Comment:
I know how design patterns work.  I disagree with the way you applied one.

This is bad encapsulation because the search algorithm is leaked into the operation view.  The operation view is a button that you click to add a step.  It shouldn't do any searching.  The palette, on the other hand, actually has a search bar and listeners for searching, so it should call the search algorithm.